File: 961.n-repeated-element-in-size-2n-array/solution.kt
class Solution {
    fun repeatedNTimes(nums: IntArray): Int {
        val v = BooleanArray(10001)
        for (n in nums) {
            if (v[n]) {
                return n
            }
            v[n] = true
        }
        throw IllegalArgumentException("No repeated element")
    }
}

File: 961.n-repeated-element-in-size-2n-array/solution.go
func repeatedNTimes(nums []int) int {
    for i := 2; i < len(nums); i++ {
        if (nums[i - 2] == nums[i] || nums[i - 1] == nums[i]) {
            return nums[i]
        }
    }
    return nums[0]
}